components/ai_chat/renderer/page_text_distilling.cc

```diff
@@ -204,4 +230,89 @@ void DistillPageText(
   std::move(callback).Run(contents_text);
 }
 
+void DistillPageTextViaSiteScript(
+    content::RenderFrame* render_frame,
+    std::string script_content,
+    int32_t world_id,
+    base::OnceCallback<void(const std::optional<std::string>&)> callback) {
+  // Concatenates our injector-script (retrieved from the resource bundle) with
+  // our extractor script. Because the injector-script is wrapped in an IIFE, we
+  // leverage custom events dispatched against the window object to communicate
+  // between the two scripts.
+  //
+  // The extractor script is currently hard-coded to request content at level 3.
+  // Level 3 is the "FULL" level, which is the highest content level. The level
+  // could be tied to the currently-active model (e.g., smaller models could be
+  // defaulted to a smaller level).
+  std::string script =
+      script_content + absl::StrFormat(
+                           R"(
```

Comment:
I'm having second thoughts about this `StrFormat` call, as everything about it looks static, but we are paying for this format every time we arrive here. I think this should be a single string constant on the unnamed namespace section of this file, and then we just append it to `script_content`. What do you think?

Reply:
This was actually a mistake on my end; the replacement should be done against the result of the two strings combined. But, I'm working on an approach to get rid of the replacements entirely, so hopefully all of this will get removed soon.

Reply:
@cdesouza-chromium the previous iteration of this did a string replace. You could split it up, but I think the benefit is very small and would make it more confusing to read

Reply:
Going to resolve this for now as the method is simpler at this point, doing only a single replacement. To @cdesouza-chromium's point though, we could definitely wrap the TypeScript output at build-time, such that we no longer need to do so here. Something for a follow-up PR, perhaps?